enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -420,78 +449,45 @@ func (s *ExecutionServer) recordExecution(
-		executionProto.RequestedTimeoutUsec = action.GetTimeout().AsDuration().Microseconds()
-
-		executionProto.RequestedComputeUnits = properties.EstimatedComputeUnits
-		executionProto.RequestedMemoryBytes = properties.EstimatedMemoryBytes
-		executionProto.RequestedMilliCpu = properties.EstimatedMilliCPU
-		executionProto.RequestedFreeDiskBytes = properties.EstimatedFreeDiskBytes
-
-		schedulingMeta := auxMeta.GetSchedulingMetadata()
-		executionProto.EstimatedFreeDiskBytes = schedulingMeta.GetTaskSize().GetEstimatedFreeDiskBytes()
-		executionProto.PreviousMeasuredMemoryBytes = schedulingMeta.GetMeasuredTaskSize().GetEstimatedMemoryBytes()
-		executionProto.PreviousMeasuredMilliCpu = schedulingMeta.GetMeasuredTaskSize().GetEstimatedMilliCpu()
-		executionProto.PreviousMeasuredFreeDiskBytes = schedulingMeta.GetMeasuredTaskSize().GetEstimatedFreeDiskBytes()
-		executionProto.PredictedMemoryBytes = schedulingMeta.GetPredictedTaskSize().GetEstimatedMemoryBytes()
-		executionProto.PredictedMilliCpu = schedulingMeta.GetPredictedTaskSize().GetEstimatedMilliCpu()
-		executionProto.PredictedFreeDiskBytes = schedulingMeta.GetPredictedTaskSize().GetEstimatedFreeDiskBytes()
-		executionProto.SelfHosted = schedulingMeta == nil || (schedulingMeta.GetExecutorGroupId() != s.env.GetSchedulerService().GetSharedExecutorPoolGroupID())
-
-		request := auxMeta.GetExecuteRequest()
-		executionProto.SkipCacheLookup = request.GetSkipCacheLookup()
-		executionProto.ExecutionPriority = request.GetExecutionPolicy().GetPriority()
-
-		regionHeaderValues := metadata.ValueFromIncomingContext(ctx, "x-buildbuddy-executor-region")
-		if len(regionHeaderValues) > 0 {
-			executionProto.Region = regionHeaderValues[len(regionHeaderValues)-1]
-		}
-
 		inv, err := s.env.GetExecutionCollector().GetInvocation(ctx, link.GetInvocationId())
 		if err != nil {
 			log.CtxErrorf(ctx, "failed to get invocation %q from ExecutionCollector: %s", link.GetInvocationId(), err)
 			continue
 		}
 		if inv == nil {
-			// The invocation hasn't finished yet. Add the execution to ExecutionCollector, and flush it once
-			// the invocation is complete
+			// The invocation hasn't finished yet. Because joins are expensive
+			// in clickhouse, we inline invocation data in the executions table.
+			// For now, add the execution to the ExecutionCollector and the build
+			// event handler will flush it after the invocation is complete.
 			if err := s.env.GetExecutionCollector().AppendExecution(ctx, link.GetInvocationId(), executionProto); err != nil {
 				log.CtxErrorf(ctx, "failed to append execution %q to invocation %q: %s", executionID, link.GetInvocationId(), err)
 			} else {
 				log.CtxInfof(ctx, "appended execution %q to invocation %q in redis", executionID, link.GetInvocationId())
 			}
 		} else if s.env.GetOLAPDBHandle() != nil {
+			// Flush to Clickhouse directly if the invocation completed before
+			// the executor published the final execution update.
```

Comment:
I think this case will become a lot more common once we start flushing to clickhouse when the stream is closed. We should probably keep an eye on clickhouse metrics in grafana to see how this is affecting disk space / compaction etc. and if it's bad enough we might want to consider some sort of batching here.

(In my local testing, I found that when doing a lot of single-row inserts, vs several large batches of inserts, the disk usage of clickhouse exploded to several GB, and it took a long time for clickhouse background merges to fully compress the data down to a few MB)

Reply:
Ack - good call out. I'll keep an eye on this